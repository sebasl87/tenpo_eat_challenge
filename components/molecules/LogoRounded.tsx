import React from 'react'
import styled from 'styled-components/native';

export interface ILogoRounded {
    image?: string;
    bgColor?: string;
}

const Container = styled.View<ILogoRounded>`
margin-top: 10px;
background-color: ${(props) => props.bgColor};
width: 104px;
height: 104px;
justify-content: center;
display: flex;
align-items: center;
border-radius: 16px;
`

const Images = styled.Image`
    max-width: 104px;
`

const LogoRounded: React.FC<ILogoRounded> = ({ image, bgColor }) => {
    return (
        <Container bgColor={bgColor}>
            <Images
                source={image}
            />
        </Container>
    )
}

export default LogoRounded;