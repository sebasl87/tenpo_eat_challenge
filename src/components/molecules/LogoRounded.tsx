import React from 'react'
import styled from 'styled-components/native';

export interface ILogoRounded {
    image?: any;
    bgColor?: string;
}

const Container = styled.View<ILogoRounded>`
    align-items: center;
    background-color: ${(props) => props.bgColor};
    border-radius: 16px;
    display: flex;
    height: 104px;
    justify-content: center;
    margin-top: 10px;
    width: 104px;
`

const Images = styled.Image`
    max-width: 104px;
`

const LogoRounded: React.FC<ILogoRounded> = ({ image, bgColor }) => {
    return (
        <Container bgColor={bgColor}>
            <Images
                source={image}
            />
        </Container>
    )
}

export default LogoRounded;