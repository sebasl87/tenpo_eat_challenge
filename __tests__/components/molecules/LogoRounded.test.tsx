import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import LogoRounded from "../../../src/components/molecules/LogoRounded";


describe("LogoRounded component", () => {
    it("should render successfully with LogoRounded", () => {
        const titleSection = renderer
            .create(<LogoRounded bgColor="#000" />)
            .toJSON();

        expect(titleSection).toMatchSnapshot();
    });
});
