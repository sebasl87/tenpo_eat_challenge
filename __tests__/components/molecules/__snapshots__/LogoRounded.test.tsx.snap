// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoRounded component should render successfully with LogoRounded 1`] = `
<View
  bgColor="#000"
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#000",
        "borderRadius": 16,
        "display": "flex",
        "height": 104,
        "justifyContent": "center",
        "marginTop": 10,
        "width": 104,
      },
    ]
  }
>
  <Image
    source={
      {
        "uri": undefined,
      }
    }
    style={
      [
        {
          "height": "100%",
          "maxWidth": 104,
          "width": "100%",
        },
      ]
    }
  />
</View>
`;
